openmdao/core/system.py

```diff
@@ -2565,3 +2608,23 @@ def _list_states(self):
             states.extend(subsys._list_states())
 
         return states
+
+    def add_recorder(self, recorder):
+        """
+        Add a recorder to the driver.
+
+        Parameters
+        ----------
+        recorder : <BaseRecorder>
+           A recorder instance.
+        """
+        self._rec_mgr.append(recorder)
+
+    def record_iteration(self):
+        """
+        Record an iteration of the current System.
+        """
+        metadata = create_local_meta(self.pathname)
+        # Send the calling method name into record_iteration, e.g. 'solve_nonlinear'.
+        self._rec_mgr.record_iteration(self, metadata, method=inspect.stack()[2][3])
```

Comment:
I *think* we put this in before we were keeping track of our own stack for the iter coord...so we may be able to get this info from our own stack now?  If not, do you know of any other ways to get the calling method name?